web/examples/craft/main.dart

```diff
@@ -93,3 +99,22 @@ void startCraft() {
     print("$fps fps, "+world.debugString());
   });
 }
+
+/// Returns the seed provided by the URL's query parameters, or null if no seed is found.
+int _getSeed() {
+  String seedQueryParam = Uri.base.queryParameters["seed"];
+  if (seedQueryParam == null) {
+    return null;
+  }
+
+  return int.tryParse(seedQueryParam);
+}
+
+/// Navigates the browser to a Url with a seed parameter and returns the seed.
+int _navigateToSeededUrl() {
+  // Note: 2^32 is the maximum value allowed.
+  int seed = new math.Random().nextInt(math.pow(2, 32));
```

Comment:
I have learned Dart is annoying in some ways, if you use `1<<32` it will assume a 32 bit integer and give you 0, so `(1<<32)-1` will give -1. What you gave works because is uses a `num` which takes advantage of the 53 bit integer in JS. I was thinking the `math.pow` here is odd since in many languages it would be a `double` but in Dart, it seems this is the best you can do without hard coding it. So this is perfect how it is, just wishing Dart had proper 64 bit integer support.

If you want to see something wonky because of Dart being stuck with JS's 53 bit integers check out [the simplex noise work around](https://github.com/Grant-Nelson/OpenSimplexNoiseDart/blob/master/lib/src/OpenSimplexNoise.dart#L29-L36) I had to do.